lib/spack/spack/package.py

```diff
@@ -983,6 +1008,124 @@ def check_paths(path_list, filetype, predicate):
                 "Install failed for %s.  Nothing was installed!" % self.name)
 
 
+    def global_license_file(self):
```

Comment:
All the stuff below here looks good, but it seems like it's bloating the already-bloated `package.py`.  I think the attributes could live in `package.py`, but maybe this stuff could go into a separate `license.py` or into a hook file.